backend/inspirehep/records/api/authors.py

```diff
@@ -97,3 +101,50 @@ def query_author_papers(bai):
 
         for data in query.yield_per(100).with_entities(RecordsAuthors.record_id):
             yield data.record_id
+
+    def update(self, data, *args, **kwargs):
+        with db.session.begin_nested():
+            super().update(data, *args, **kwargs)
+            self.update_students_advisors_table()
+
+    def get_linked_advisors_when_name_changes(self):
+        if not self.get("advisors"):
+            return set()
+        if (
+            (
+                self.get_value("name.preferred_name")
+                and self.get_value("name.preferred_name")
+                != self._previous_version.get_value("name.preferred_name")
+            )
+            or (
+                self.get_value("name.preferred_name")
+                != self._previous_version.get_value("name.preferred_name")
+            )
+            or (
+                self.get_value("name.value")
+                != self._previous_version.get_value("name.value")
+            )
```

Comment:
* the second clause is now a superset of the first one
* it's a bit too broad (`name.value` changes only matter if there's no `preferred_name`), however I don't think that really matters. We'll be reindexing advisors a little bit too frequently in some cases.